src/libraries/Common/src/Interop/BSD/System.Native/Interop.Sysctl.cs

```diff
@@ -48,8 +49,33 @@ private static unsafe void Sysctl(int* name, int name_len, ref byte* value, ref
             }
 
             ret = Sysctl(name, name_len, value, &bytesLength);
+            while (autoSize && ret != 0 && GetLastErrorInfo().Error == Error.ENOMEM)
+            {
+                // Do not use ReAllocHGlobal() here: we don't care about
+                // previous contents, and proper checking of value returned
+                // will make code more complex.
+                Marshal.FreeHGlobal((IntPtr)value);
+                if ((int)bytesLength == int.MaxValue)
```

Comment:
I'm not sure why the int.MaxValue has significance. Since bytesLength is IntPtr we can possibly allocate more than that even if MaxValue is already pretty big. It feels like we can simply use 2x and let Marshal.AllocHGlobal to decide if there is memory or not.

Reply:
Technically, `int.MaxValue` is more than enough: `sysctl(3)` interface was never designed to pass large amounts of data. In the worst case I know (list of all processes and threads in system, or list of all routes on a BGP router) there will be only a few megabytes of data incoming.

The `Sys.Sysctl()` interface specifies `ref int len` already, and changing that would be a different issue anyway.